components/applications-service/pkg/wrap/wrap_test.go

```diff
@@ -0,0 +1,51 @@
+package wrap
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/stretchr/testify/require"
+
+	"github.com/chef/automate/api/external/applications"
+)
+
+func TestWrap(t *testing.T) {
+	ex1 := applications.ExampleOne{
+		IntOne: 23,
+		IntTwo: 42,
+	}
+
+	packedEx1, err := ptypes.MarshalAny(&ex1)
+	require.NoError(t, err)
+
+	wrap1 := applications.Wrapper{
+		Type:    "ExampleOne",
+		Message: packedEx1,
+	}
+
+	fmt.Printf("w: %+v\n", wrap1)
```

Comment:
If you look at the output of this, it adds some fixed amount of overhead we have to deal with. The output is like:

```
{Type:ExampleOne Message:type_url:"type.googleapis.com/chef.automate.api.applications.ExampleOne" value:"\010\027\020*"  XXX_NoUnkeyedLiteral:{} XXX_unrecognized:[] XXX_sizecache:0}
```

So we have stored our own definition of what the type is, and then `ptypes.MarshalAny()` has also converted our message to a thing with its own definition of what the type is. It might be possible to remove our type "tag" and use the one added by protobuf.Any, see https://github.com/golang/protobuf/blob/e91709a02e0e8ff8b86b7aa913fdc9ae9498e825/ptypes/any/any.pb.go#L104-L129

Reply:
So one alternative would be to make the wrapper like so (proto):

```protobuf
message Wrapper {
  string type = 1;
  bytes message = 2;
}
```

This is what `protobuf.Any` does internally anyway: https://github.com/golang/protobuf/blob/e91709a02e0e8ff8b86b7aa913fdc9ae9498e825/ptypes/any/any.proto#L122-L154

The advantages of rolling our own would be:
* shorter type annotations in the message for less message size overhead
* wire compatibility is not broken if we change the protobuf package name(space)